src/crypto/crypto_base.ts

```diff
@@ -0,0 +1,4 @@
+export abstract class CryptoBase {
```

Comment:
Nit: This should probably be an interface instead of an abstract class since it has no implementation. Additionally, we should probably make sure that `ActionCrypto` is typed with the interface `CryptoBase` versus having the concrete type accessible to actions. Also maybe we should rename this something more like `CryptoProvider`.